build.gradle.kts

```diff
@@ -0,0 +1,99 @@
+import java.net.URL
+
+plugins {
+    id("com.hivemq.extension")
+    id("com.github.hierynomus.license")
+    id("io.github.sgtsilvio.gradle.defaults")
+    id("org.asciidoctor.jvm.convert")
+}
+
+group = "com.hivemq.extensions"
+description = "HiveMQ Extension for cluster discovery with S3 buckets"
+
+hivemqExtension {
+    name.set("S3 Cluster Discovery Extension")
+    author.set("HiveMQ")
+    priority.set(1000)
+    startPriority.set(10000)
+    sdkVersion.set("${property("hivemq-extension-sdk.version")}")
+}
+
+dependencies {
+    hivemqProvided("ch.qos.logback:logback-classic:${property("logback.version")}")
+    implementation("org.aeonbits.owner:owner:${property("owner.version")}")
+    implementation(platform("software.amazon.awssdk:bom:${property("aws-bom.version")}"))
+    implementation("software.amazon.awssdk:s3")
+}
+
+/* ******************** resources ******************** */
+
+val prepareAsciidoc by tasks.registering(Sync::class) {
+    from("README.adoc").into({ temporaryDir })
+}
+
+tasks.asciidoctor {
+    dependsOn(prepareAsciidoc)
+    sourceDir(prepareAsciidoc.map { it.destinationDir })
+}
+
+hivemqExtension.resources {
+    from("LICENSE")
+    from("README.adoc") { rename { "README.txt" } }
+    from(tasks.asciidoctor)
+}
+
+/* ******************** test ******************** */
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter:${property("junit-jupiter.version")}")
+    testImplementation("org.mockito:mockito-core:${property("mockito.version")}")
+    testImplementation("org.mockito:mockito-junit-jupiter:${property("mockito.version")}")
+}
+
+tasks.withType<Test>().configureEach {
+    useJUnitPlatform()
+}
+
+/* ******************** integration test ******************** */
+
+dependencies {
+    //necessary as the localstack s3 service would not start without the old sdk
+    integrationTestImplementation("com.amazonaws:aws-java-sdk-s3:${property("aws-legacy-sdk.version")}")
```

Comment:
Can this be a runtime dependency then? This could be confusing on imports, when writing tests.

Reply:
Good point.